var/spack/repos/builtin/packages/py-ipython/package.py

```diff
@@ -35,31 +24,51 @@ class PyIpython(PythonPackage):
+    depends_on('py-jedi@0.16:', when='@7.18,7.20:', type=('build', 'run'))
+    depends_on('py-jedi@0.10:', when='@7.5:7.17,7.19', type=('build', 'run'))
+    depends_on('py-black', when='@8:', type=('build', 'run'))
     depends_on('py-decorator', type=('build', 'run'))
     depends_on('py-pickleshare', type=('build', 'run'))
+    depends_on('py-traitlets@5:', when='@8:', type=('build', 'run'))
     depends_on('py-traitlets@4.2:', type=('build', 'run'))
-    depends_on('py-prompt-toolkit@1.0.4:1',  when='@:7.0.0', type=('build', 'run'))
-    depends_on('py-prompt-toolkit@2.0.0:2',  when='@7.0.0:7.5.0', type=('build', 'run'))
-    depends_on('py-prompt-toolkit@2.0.0:2.0', when='@7.5.0', type=('build', 'run'))
-    depends_on('py-prompt-toolkit@3.0.2:3.0', when='@7.18:7.25', type=('build', 'run'))
     depends_on('py-prompt-toolkit@2.0.0:2,3.0.2:3.0', when='@7.26:', type=('build', 'run'))
+    depends_on('py-prompt-toolkit@3.0.2:3.0', when='@7.18:7.25', type=('build', 'run'))
+    depends_on('py-prompt-toolkit@2.0.0:2.0', when='@7.5.0', type=('build', 'run'))
+    depends_on('py-prompt-toolkit@2.0.0:2', when='@7.0.0:7.5.0', type=('build', 'run'))
+    depends_on('py-prompt-toolkit@1.0.4:1', when='@:7.0.0', type=('build', 'run'))
     depends_on('py-pygments', type=('build', 'run'))
-    depends_on('py-backcall', type=('build', 'run'), when='@7.3.0:')
+    depends_on('py-backcall', when='@7.3.0:', type=('build', 'run'))
+    depends_on('py-stack-data', when='@8:', type=('build', 'run'))
     depends_on('py-matplotlib-inline', when='@7.23:', type=('build', 'run'))
+    depends_on('py-pexpect@4.4:', when='@7.18: platform=linux', type=('build', 'run'))
+    depends_on('py-pexpect@4.4:', when='@7.18: platform=darwin', type=('build', 'run'))
+    depends_on('py-pexpect@4.4:', when='@7.18: platform=cray', type=('build', 'run'))
     depends_on('py-pexpect', type=('build', 'run'))
-    depends_on('py-pexpect@4.4:', type=('build', 'run'), when='@7.18: platform=linux')
-    depends_on('py-pexpect@4.4:', type=('build', 'run'), when='@7.18: platform=darwin')
-    depends_on('py-pexpect@4.4:', type=('build', 'run'), when='@7.18: platform=cray')
-    depends_on('py-appnope', type=('build', 'run'), when='platform=darwin')
+    depends_on('py-appnope', when='platform=darwin', type=('build', 'run'))
     depends_on('py-colorama', when='platform=windows', type=('build', 'run'))
-    depends_on('py-backports-shutil-get-terminal-size', type=('build', 'run'), when="^python@:3.2")
-    depends_on('py-pathlib2', type=('build', 'run'), when="^python@:3.3")
-    depends_on('py-simplegeneric@0.8:', type=('build', 'run'), when='@:7.0.0')
+
+    depends_on('py-backports-shutil-get-terminal-size', when="^python@:3.2", type=('build', 'run'))
+    depends_on('py-pathlib2', when="^python@:3.3", type=('build', 'run'))
+    depends_on('py-simplegeneric@0.8:', when='@:7.0.0', type=('build', 'run'))
+
+    @property
+    def import_modules(self):
```

Comment:
We should really add a "skip_modules" attribute to `PythonPackage` that does this same thing. I've been meaning to do that for a while. Feel free to open an issue to remind me of this or open a PR to implement it if you're bored. Anyway, this solution is fine for now.